### DISCUSSION TITLE
Java version set but not working

### DISCUSSION THREAD

**Discussion by Blast06** [codemagic-ci-cd/codemagic-docs#2206] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1.Set a java version on a project(according to the docs)
2. run the workflow
3. Wait for the results in the first step of the workflow

This is how i added the variable
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/20145023/682d3624-b025-4443-93f9-4a7953b32d08)

The output:

== Use Xcode 15 ==
> sudo xcode-select --switch /Applications/Xcode-15.0.app/Contents/Developer
Xcode 15.0.1
Build version 15A507

== Use default version of CocoaPods ==
> gem list '^cocoapods$' --no-verbose
cocoapods (1.13.0)

== JAVA_HOME set to /usr/lib/jvm/java-1.17.0-openjdk-amd64 ==

Build failed :|
Invalid JAVA_HOME path set. Check our documentation for the correct paths.

### Expected results

Compiled project successfully

### Actual results

Not Compiling project successfully

### Build id (optional)

655919bfcd7f6dbdd9f6213f

**Comment by himesh-cm** [comment, marked as answer]
Hi @Blast06, Please check the supported `java version` [here](https://docs.codemagic.io/specs/versions-macos/) in the docs as per the build machine specified in your workflow
